app/src/org/commcare/activities/HomeScreenBaseActivity.java

```diff
@@ -392,7 +391,7 @@ protected void showLocaleChangeMenu(final CommCareActivityUIController uiControl
                 Localization.setLocale("default");
             } else {
                 String selectedLocale = localeCodes[position];
-                MainConfigurablePreferences.setCurrentLocale(selectedLocale);
+                LocalePreferences.saveCurrentLocale(selectedLocale);
```

Comment:
Sorry for not making it clear upfront we don't want to use CommCare language settings for switching the Layout to RTL. Instead we wanna rely on System locale (Phone locale settings) only to make this switch. The reason being we don't want to be consistent with rest of the phone in choosing the layout direction.

Reply:
Could we continue this discussion on trello? I've received different instructions from A.King.